### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by caleybrock**
Updating documentation to prepare for the project handoff.

- reviewed and updated all readmes
- reviewed and updated all /docs
- added diagrams and screenshots
- add sections on CI/CD, the tech stack, docker setup, testing, product

### CODE REVIEW

File: TESTING.md [hunk 1]
```diff
@@ -15,3 +17,33 @@ In powershell:
...
+
+In powershell:
+
+```shell
+docker exec -it project-thundercat_backend_1 /bin/bash
+
+./manage.py test
+```
+
+TODO(fnormand01) - Outline how to run Postman tests.
```
Comment (caleybrock) [inline TESTING.md]: @fnormand01 could you fill this in?

File: backend/README.md [hunk 2]
```diff
@@ -10,29 +10,36 @@ In the project directory, you can run the following commands.
...
 
-Applies migrations that haven't been migrated yet.<br><br>
+### Scripts
 
-### `python manage.py showmigrations backend`
+You'll have to login to the backend container to run these scripts: `docker exec -it project-thundercat_backend_1 /bin/bash`
 
-Lists project's migrations and their status.<br><br>
+Create new migrations based on the changes you've made to your models:
+`python manage.py makemigrations backend`
```
Comment (fnormand01) [inline backend/README.md]: We're now using `custom_models` instead of `backend`.

File: backend/README.md [hunk 3]
```diff
@@ -10,29 +10,36 @@ In the project directory, you can run the following commands.
...
 
-### `python manage.py migrate <migration_name> backend`
+Apply migrations that haven't been migrated yet:
+`python manage.py migrate backend`
```
Comment (fnormand01) [inline backend/README.md]: Here, we need to do:
`python manage.py migrate` OR `python manage.py migrate custom_models` if we want to migrate only what's under _custom_models_.

File: backend/README.md [hunk 4]
```diff
@@ -10,29 +10,36 @@ In the project directory, you can run the following commands.
...
 
-Migrates to a previous specified version.<br><br>
+List project's migrations and their status:
+`python manage.py showmigrations backend`
```
Comment (fnormand01) [inline backend/README.md]: Here, if we want to show all project migrations:
`python manage.py showmigrations`

If we want to show only our custom migrations status:
`python manage.py showmigrations custom_models`

File: backend/README.md [hunk 5]
```diff
@@ -10,29 +10,36 @@ In the project directory, you can run the following commands.
...
 
-See [Django Migrations](https://docs.djangoproject.com/en/2.1/topics/migrations/) for more information.
\ No newline at end of file
+Migrate to a previous specified version:
+`python manage.py migrate <migration_name> backend`
```
Comment (fnormand01) [inline backend/README.md]: To migrate to previous versions, we must do as followed:
`python manage.py migrate custom_models <migration_name>`

File: backend/README.md [hunk 6]
```diff
@@ -10,29 +10,36 @@ In the project directory, you can run the following commands.
...
+
+To rollback all migrations and re-run them:
```
Comment (fnormand01) [inline backend/README.md]: Here, I would specify that it is rolling back all migration under _custom_models_ only using the following command.

File: TESTING.md [hunk 7]
```diff
@@ -15,3 +17,33 @@ In powershell:
...
+The backend Django application has a set of tests for it's views and uses Postman to manage API tests.
+
+To run the unit tests, you'll need to enter the frontend Docker container.
+
+In powershell:
+
+```shell
+docker exec -it project-thundercat_backend_1 /bin/bash
+
+./manage.py test
```
Comment (fnormand01) [inline TESTING.md]: Here, we need to add `tests`, like that:
`./manage.py test tests`

File: TESTING.md [hunk 8]
```diff
@@ -15,3 +17,33 @@ In powershell:
...
+docker exec -it project-thundercat_frontend_1 /bin/bash
+
+npm run test
+```
+
+## Backend tests
+
+The backend Django application has a set of tests for it's views and uses Postman to manage API tests.
+
+To run the unit tests, you'll need to enter the frontend Docker container.
```
Comment (fnormand01) [inline TESTING.md]: Should be _"backend Docker container"_ instead of _"frontend Docker container"_.

File: TESTING.md (hunk 1 above)
Comment (fnormand01) [reply TESTING.md]: @caleybrock I added some postman documentation.
See commit #https://github.com/code-for-canada/project-thundercat/pull/475/commits/fdef200a012447c8290b304a7775368faf3c88c7

**Pull request merged**